extensions/typescript-language-features/src/typescriptServiceClient.ts

```diff
@@ -470,6 +470,46 @@ export default class TypeScriptServiceClient extends Disposable implements IType
 		this._versionManager.promptUserForVersion();
 	}
 
+	public async openTsServerTraceDirectory(): Promise<boolean> {
+		if (!this._configuration.enableTsServerTracing) {
+			vscode.window.showErrorMessage<vscode.MessageItem>(
+				localize(
+					'typescript.openTsServerTrace.tracingNotEnabled',
+					'TS Server tracing is off. Please set `typescript.tsserver.enableTracing` and restart the TS server to enable logging'),
+				{
+					title: localize(
+						'typescript.openTsServerTrace.enableAndReloadOption',
+						'Enable tracing and restart TS server'),
+				})
+				.then(selection => {
+					if (selection) {
+						return vscode.workspace.getConfiguration().update('typescript.tsserver.enableTracing', true, true).then(() => {
+							this.restartTsServer();
+						});
+					}
+					return undefined;
+				});
+			return false;
+		}
+
+		if (this.serverState.type !== ServerState.Type.Running || !this.serverState.server.tsServerTraceDirectory) {
+			vscode.window.showWarningMessage(localize(
+				'typescript.openTsServerTrace.noTraceDirectory',
+				'TS Server has not started tracing.'));
+			return false;
+		}
+
+		try {
+			await vscode.commands.executeCommand('revealFileInOS', vscode.Uri.file(this.serverState.server.tsServerTraceDirectory));
```

Comment:
I copied this from the server log handler, but it doesn't actually work.  The command ignores its argument and reveals the current file (and only if a folder is open).  Rather than fix it, I made the PR a draft until I get confirmation that we actually want a command (which I doubt).

Reply:
Let's start by just logging the server trace directory instead of adding a command.

I imagine we'll usually be asking user to go through a set of steps to collect server traces. If we find that getting to the trace file itself is confusing, we revisit adding a command to simplify things